src/gui/src/gui.cpp

```diff
@@ -431,97 +431,108 @@ void Gui::deleteRuler(const std::string& name)
                 const std::string& name_filter,
                 const std::string& attribute,
                 const std::any& value,
                 bool filter_case_sensitive,
                 int highlight_group)
 {
-  for (auto& [object_type, descriptor] : descriptors_) {
-    if (descriptor->getTypeName() == type) {
-      SelectionSet selected_set;
-      if (descriptor->getAllObjects(selected_set)) {
-        if (!name_filter.empty()) {
-          // convert to vector
-          std::vector<Selected> selected_vector(selected_set.begin(),
-                                                selected_set.end());
-          // remove elements
 #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
-          // Define case sensitivity options for QRegularExpression
-          QRegularExpression::PatternOptions options
-              = filter_case_sensitive
-                    ? QRegularExpression::NoPatternOption
-                    : QRegularExpression::CaseInsensitiveOption;
-
-          // Convert the wildcard string to a regex pattern and create the
-          // object
-          QRegularExpression reg_filter(
-              QRegularExpression::wildcardToRegularExpression(
-                  QString::fromStdString(name_filter)),
-              options);
+  // Define case sensitivity options for QRegularExpression
+  const QRegularExpression::PatternOptions options
+      = filter_case_sensitive ? QRegularExpression::NoPatternOption
+                              : QRegularExpression::CaseInsensitiveOption;
+
+  // Convert the wildcard string to a regex pattern and create the
+  // object
+  const QRegularExpression reg_filter(
+      QRegularExpression::wildcardToRegularExpression(
+          QString::fromStdString(name_filter)),
+      options);
 #else
-          QRegExp reg_filter(
-              QString::fromStdString(name_filter),
-              filter_case_sensitive ? Qt::CaseSensitive : Qt::CaseInsensitive,
-              QRegExp::WildcardUnix);
+  const QRegExp reg_filter(
+      QString::fromStdString(name_filter),
+      filter_case_sensitive ? Qt::CaseSensitive : Qt::CaseInsensitive,
+      QRegExp::WildcardUnix);
 #endif
-          auto remove_if = std::remove_if(
-              selected_vector.begin(),
-              selected_vector.end(),
-              [&name_filter, &reg_filter](auto sel) -> bool {
-                const std::string sel_name = sel.getName();
-                if (sel_name == name_filter) {
-                  // direct match, so don't remove
-                  return false;
-                }
+  const bool is_simple = isSimpleStringPattern(name_filter);
+  for (auto& [object_type, descriptor] : descriptors_) {
+    if (descriptor->getTypeName() != type) {
+      continue;
+    }
+    SelectionSet selected_set;
+    descriptor->visitAllObjects([&](const Selected& sel) {
+      if (!name_filter.empty()) {
+        const std::string sel_name = sel.getName();
+        if (sel_name != name_filter
```

Comment:
if is_simple is False, it this string compare needed?

Reply:
Good point